src/IO/tcp/SocketClient.cpp

```diff
@@ -0,0 +1,65 @@
+#include "SocketClient.h"
+
+#include <mutex>
+#include <spdlog/spdlog.h>
+#include <thread>
+
+using namespace boost::asio;
+
+SocketClient::SocketClient(const std::shared_ptr<ip::tcp::socket> socket, 
+        std::function<void(const char b[])> receiveCallback,
+        std::function<void(SocketClient*)> closeCallback):
+    socket(socket),
+    receiveCallback(receiveCallback),
+    closeCallback(closeCallback) 
```

Comment:
Clang Tidy recommends this:
```suggestion
SocketClient::SocketClient(std::shared_ptr<ip::tcp::socket> socket, 
        std::function<void(const char[])> receiveCallback,
        std::function<void(SocketClient*)> closeCallback):
    socket(std::move(socket)),
    receiveCallback(std::move(receiveCallback)),
    closeCallback(std::move(closeCallback)) 
```

Reply:
Will move cause the shared pointer to be copied instead of adding a new reference (and increasing the reference count)?

Reply:
From what I understand, you will have one copy when you pass the shared_pointer to the `SocketClient` constructor. This one you need to have. If you don't use `std::move` here, you would have a second copy when assigning it to your socket (and then afterward the first copy will get out of scope). However, I think `std::move` prevents that second copy.